src/App.jsx

```diff
@@ -20,20 +40,28 @@ const App = () => {
   const addChip = (id) => {
     const newDogs = dogs.map((dog) => {
       if (dog.id === id) {
-        return {
-          ...dog,
-          chip: String(getRndInteger(1000, 9999)),
-        };
+        dog.chip = String(getRndInteger(1000, 9999));
+        axios
+          .patch(`${URL}/${id}/add_chip`)
+          .then(() => setDogs(newDogs))
+          .catch((error) => console.log(error));
       }
       return dog;
     });
-    setDogs(newDogs);
   };
```

Comment:
The flow of this code is surprising. I would expect it to make the one patch api call for the dog of interest, then use that result to replace the dog record in the local dog data. Something more like this
```suggestion
  const addChip = (id) => {
    axios
      .patch(`${URL}/${id}/add_chip`)
      .then(response => {
        const newDog = response.data;
        const newDogs = dogs.map(dog => {
          if (dog.id === newDog.id) {
            return newDog;
          }
          return dog;
        });
        setDogs(newDogs);
      })
      .catch((error) => console.log(error));
  };
```